[Do not merge] dApp-friendly contract deployment

r-czajkowski opened the pull request:
Ref: https://github.com/threshold-network/token-dashboard/issues/136

We need a dApp-friendly version of deployment allowing dashboard developers to build UI components without having to run geth/ganache/hardhat locally and deploying contracts locally.

This PR aims at presenting changes to smart contracts deployed to Sepolia that are used by the dApp development team for day-to-day work. Please do not merge this PR to `main` branch. Smart contract with the changes presented here are pushed to NPM registry with a tag `dapp-development-sepolia`.

**Main changes:**
* changes the `authorizationDecreaseDelay` and `authorizationDecreaseChangePeriod` to `180 seconds` instead of `45 days`, so we can test the authorization decrease flow faster in the T dapp on testnet.

r-czajkowski committed:
Fine-tuned authorization parameters

Change the `authorizationDecreaseDelay` and
`authorizationDecreaseChangePeriod` to `180` seconds in `RandomBeacon`
and `WalletRegistry`. It means that we can test the authroziation
decrease flow faster in the T dapp on testnet.

michalinacienciala commented:
Could you change this PR to draft and add a visible info in the description that those changes are not supposed to get merged to `main`?
Also, I was wondering if there's some way to additionally block the merging of this PR via GH settings. I mean something similar to protected branches functionality. From a quick googling I don't see such setting... Marking the PR as draft and explaining that the changes should not go to `main` will need to be enough then...

pdyraga commented on CODEOWNERS:
```diff
@@ -31,3 +31,6 @@
 /solidity-v1/contracts/utils/UIntArrayUtils.sol @Shadowfiend @pdyraga
 /solidity-v1/package.json @Shadowfiend @pdyraga
+
+/solidity/random-beacon/contracts/RandomBeacon.sol @pdyraga @nkuba @lukasz-zimnoch
+/solidity/ecdsa/contracts/WalletRegistry.sol @pdyraga @nkuba @lukasz-zimnoch
```
Let's extract `CODEOWNERS` change to a separate PR and merge it to `main`. 

The `CODEOWNERS` should cover two contracts listed here as well as all files in:
- `solidity/ecdsa/contracts/libraries` 
- `solidity/ecdsa/contracts/api`
- `solidity/random-beacon/contracts/libraries/`
- `solidity/random-beacon/contracts/api`

Let's add there all developers who worked on the Solidity code of the mentioned smart contracts: `@pdyraga @nkuba @lukasz-zimnoch @dimpar @tomaszslabon`.

This will not only protect dApp-dev-only changes from being merged but will also protect from any accidental changes to an already audited code.

r-czajkowski replied:
https://github.com/keep-network/keep-core/pull/3125

r-czajkowski committed:
Merge branch 'main' into dapp-development

r-czajkowski committed:
Add a mock function that creates wallet

We need to create the mocked wallet in this contract because the
tbtc-v2.ts lib looks up wallet by id to get the active wallet public
key. Please see
https://github.com/keep-network/tbtc-v2/blob/main/typescript/src/ethereum.ts#L392-L395.
The `Bridge` contract from the `tbtc-v2` stores the hash of the wllet's
public key not public key.

michalinacienciala commented:
@r-czajkowski, when I tested the deployment of `ecdsa` contracts from `dapp-development` branch, I got following error:
```
································|··············
 |  WalletRegistryGovernance     ·     17.088  │
 ································|··············
 |  Wallets                      ·      0.086  │
 ·-------------------------------|-------------·

Error in plugin hardhat-contract-sizer: Warning: 1 contracts exceed the size limit for mainnet deployment.

For more info run Hardhat with --show-stack-traces
error Command failed with exit code 1.
```

This happens both on `yarn deploy:test` and `yarn build`.

r-czajkowski committed:
Merge branch 'main' into dapp-development

nkuba commented on solidity/ecdsa/contracts/WalletRegistry.sol:
```diff
@@ -750,6 +750,16 @@ contract WalletRegistry is
+    ) external {
+        wallets.registry[ecdsaWalletID].publicKeyX = publicKeyX;
+        wallets.registry[ecdsaWalletID].publicKeyY = publicKeyY;
+        emit WalletCreated(ecdsaWalletID, 0x0);
```
Maybe use some dummy value instead of `0x0` which seems a default empty value?

r-czajkowski committed:
Get rid of an event from mocked function

After merging the `main` branch we hit the `WalletRegisrty` contract
size limit. So here we get rid of the `WalletCreated` event to reduce
the contract size.

r-czajkowski committed:
Update contract deps

Set `dapp-development-goerli` tag for `@keep-network/random-beacon` and
`@threshold-network/solidity-contracts`. We need dapp-friendly contracts
here.

michalinacienciala committed:
Merge remote-tracking branch 'origin/main' into dapp-development

michalinacienciala committed:
Merge remote-tracking branch 'origin/main' into dapp-development

michalinacienciala committed:
Change version's suffix to `-dapp-dev-goerli`

We are publishing packages with code from `dapp-development` branch under
versions that use `-dapp-dev-goerli.X` suffix. Our `package.json` on this branch
should reflect that. If we do not set it up, the CI job used to pubblish the
package will not be able to correctly bump up the version of the package.

michalinacienciala committed:
Update contract deps

Set `dapp-development-goerli` tag for `@threshold-network/solidity-contracts`.
We need dapp-friendly contracts here.

r-czajkowski committed:
Merge branch 'main' into dapp-development

r-czajkowski committed:
Merge branch 'main' into dapp-development

r-czajkowski committed:
Update `yarn.lock` in random beacon package

We need the latest version of the `@threshold-network/solidity-contrcts`
with the `dapp-development-goerli` tag.

r-czajkowski committed:
Update `yarn.lock` in ecdsa package

We need the latest version of the
`@threshold-network/solidity-contracts` and `keep-network/random-beacon`
with the `dapp-development-goerli` tag.

r-czajkowski committed:
Remove `tenderly` tag for `goerli` network

We changed the version range for `@tenderly/hardhat-tenderly` in
https://github.com/keep-network/keep-core/pull/3454 (ecdsa package) and
CI build fails on tenderly verification. The tenderly verification
passes in random-beacon package because we use a fixed `1.0.12` version.
Since this is a `dapp-development` branch, we do not actually need the
tenderly verification at least for now so here we remove the `tenderly`
tag for `goerli` network to skip the tenderly verification.

michalinacienciala committed:
Merge remote-tracking branch 'origin/main' into dapp-development

michalinacienciala committed:
Switch to dapp-dev-sepolia

We're switching deployment to the Sepolia testnet due to planned deprecation of
the Goerli testnet. In a previous commit we've updated the `dapp-development`
branch with the recent changes from `main` (among them were the changes adding
Sepolia to the list of supported networks). Now we're updating the
`dapp-development` branch to create `dapp-dev-sepolia` packages.

michalinacienciala committed:
Switch to dapp-dev-sepolia

We're switching deployment to the Sepolia testnet due to planned deprecation of
the Goerli testnet. In a previous commit we've updated the `dapp-development`
branch with the recent changes from `main` (among them were the changes adding
Sepolia to the list of supported networks). Now we're updating the
`dapp-development` branch to create `dapp-dev-sepolia` packages.
